pkg/plugins/manager/pipeline/initialization/initialization.go

```diff
@@ -0,0 +1,67 @@
+package initialization
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/config"
+	"github.com/grafana/grafana/pkg/plugins/log"
+)
+
+// Initializer is responsible for the Initialization stage of the plugin loader pipeline.
+type Initializer interface {
+	Initialize(ctx context.Context, ps []*plugins.Plugin) ([]*plugins.Plugin, error)
+}
+
+// InitializeFunc is the function used for the Initialize step of the Initialization stage.
+type InitializeFunc func(ctx context.Context, p *plugins.Plugin) (*plugins.Plugin, error)
+
+type Initialize struct {
+	cfg             *config.Cfg
+	initializeSteps []InitializeFunc
+	log             log.Logger
+}
+
+type Opts struct {
+	InitializeFuncs []InitializeFunc
+}
+
+// New returns a new Initialization stage.
+func New(cfg *config.Cfg, opts Opts) *Initialize {
+	if len(opts.InitializeFuncs) == 0 {
```

Comment:
nit: the check here can be `if opts.InitializeFuncs == nil`. Also, can we name this only *Funcs or *Steps?

Reply:
Good point - we probably want to accept the input list regardless if default empty/non-empty. 

RE the naming, I think for consistency with the other stages it's best to keep as `InitializeFuncs` for now - but see your point. I'm not even entirely convinced on the `Initialization` stage naming here, but we can worry about that later once all stages are in place I think